### PR TITLE
Qt/ui: update vipc in thread

### LINKED ISSUE #22257: C3: UI state updates at 10Hz
Camera view seems fine, but model and other state only updates at 10Hz

### PR DESCRIPTION
**Pull request by deanlee**
fix UI state updates at 10Hz while onroad. resolved  #22257

### CODE REVIEW

File: selfdrive/ui/ui.h [hunk 1]
```diff
@@ -117,7 +117,9 @@ typedef struct UIScene {
   uint64_t started_frame;
 } UIScene;
 
+class QUIState;
```
Comment (pd0wm) [inline selfdrive/ui/ui.h:120]: Can we just remove the `typedef struct`? That's a leftover from the original c days.

**Comment by adeebshihadeh** [comment]
Can we move the camera view updating to another thread instead?

**Comment by deanlee** [comment]
moved vipc to thread.it doesn't block the main thread now.

**Comment by deanlee** [comment]
need to stop "try to connect or recv again" when offroad. otherwise it will  loop infinite

**Comment by deanlee** [comment]
there is a resource leak in VisionIpcClient, fixed in   https://github.com/commaai/cereal/pull/201

**Comment by deanlee** [comment]
moved to #22354

**Pull request closed without merging**